### PR TITLE
Thinkmd library integration

### PR DESCRIPTION
**Pull request by zzainulabidin**

### CODE REVIEW

File: opensrp-chw/src/main/java/org/smartregister/chw/dao/VisitDao.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package org.smartregister.chw.dao;
+
+import org.smartregister.dao.AbstractDao;
+
+import java.util.List;
+
+public class VisitDao extends AbstractDao {
```
Comment (githengi) [inline opensrp-chw/src/main/java/org/smartregister/chw/dao/VisitDao.java:7]: Move this to chw-core library

File: opensrp-chw/build.gradle [hunk 2]
```diff
@@ -51,6 +51,12 @@ android {
         buildConfigField "int", "MAX_SYNC_RETRIES", '3'
         buildConfigField "long", "BUILD_TIMESTAMP", System.currentTimeMillis() + "L"
         buildConfigField "boolean", "IS_SYNC_SETTINGS", "false"
+        buildConfigField "String", "THINKMD_BASE_URL", '"https://thinkmd-v4-app-b-ona.herokuapp.com/"'
+        buildConfigField "String", "THINKMD_END_POINT", '"https://fhir-v4-staging-thinkmd-tech.herokuapp.com/fhir"'
+        buildConfigField "String", "PATIENT_IDENTIFIER", '"https://fhir.smartregister.org/thinkMD_id"'
```
Comment (githengi) [inline opensrp-chw/build.gradle:56]: I think will not be changing per thinkmd integration and should be defined as constants on `ThinkMDConfig`

File: opensrp-chw/build.gradle [hunk 3]
```diff
@@ -51,6 +51,12 @@ android {
...
+        buildConfigField "String", "ENCOUNTER_IDENTIFIER", '"https://fhir.smartregister.org/encounterUuid"'
+        buildConfigField "String", "MESSAGE_EVENT_URL", '"http://hl7.org/fhir/ValueSet/message-events"'
+        buildConfigField "String", "PRACTITIONER_COMMUNICATION_CODING_SYSTEM", '"urn:ietf:bcp:47"'
```
Comment (githengi) [inline opensrp-chw/build.gradle:59]: I think will not be changing per thinkmd integration and should be defined as constants on `ThinkMDConfig`

File: opensrp-chw/src/main/java/org/smartregister/chw/util/Utils.java [hunk 4]
```diff
@@ -140,4 +155,48 @@ public static void updateToolbarTitle(Activity activity, int toolbarTextViewId)
...
+        }
+        return null;
+    }
+
+    public static Pair<String, String> fetchMUACFromDB(String childBaseEntityId) {
+        String muacValue = org.smartregister.chw.dao.VisitDao.getMUACValue(childBaseEntityId);
+        String muacCode = muacValue.substring(4);
+        String muacDiaplay = muacCode.substring(0, 1).toUpperCase() + muacCode.substring(1);
+        return Pair.create(muacCode, muacDiaplay);
+    }
```
Comment (githengi) [inline opensrp-chw/src/main/java/org/smartregister/chw/util/Utils.java:199]: Move to chw-core library as a DAO class

File: opensrp-chw/src/main/java/org/smartregister/chw/activity/ChildProfileActivity.java [hunk 5]
```diff
@@ -150,14 +159,46 @@ public boolean onOptionsItemSelected(MenuItem item) {
...
+        //Todo: these values needs to be query and set into model
+        model.setPractitionerId(null);
+        model.setPatientId(null);
+        model.setUserName(null);
+        model.setLocationId(null);
+        model.setUniqueIdGeneratedForThinkMD(null);
+        model.setEndPointPackageName(null);
+
+        return model;
+    }
```
Comment (githengi) [inline opensrp-chw/src/main/java/org/smartregister/chw/activity/ChildProfileActivity.java:200]: Define this in a class as a method that be overriden by a CHW flavour. You also need to pass the `childBaseEntityId` to the method

Comment (githengi) [reply opensrp-chw/src/main/java/org/smartregister/chw/activity/ChildProfileActivity.java:200]: The base Implementation can be moved also to CHW core library

File: opensrp-chw/build.gradle (hunk 3 above)
Comment (zzainulabidin) [reply opensrp-chw/build.gradle:59]: On ThinkMDConfig we have private variables with getters and setters.

File: opensrp-chw/src/main/java/org/smartregister/chw/dao/VisitDao.java (hunk 1 above)
Comment (zzainulabidin) [reply opensrp-chw/src/main/java/org/smartregister/chw/dao/VisitDao.java:7]: Done

File: opensrp-chw/src/main/java/org/smartregister/chw/util/Utils.java (hunk 4 above)
Comment (zzainulabidin) [reply opensrp-chw/src/main/java/org/smartregister/chw/util/Utils.java:199]: Done

File: opensrp-chw/src/main/java/org/smartregister/chw/activity/ChildProfileActivity.java (hunk 5 above)
Comment (zzainulabidin) [reply opensrp-chw/src/main/java/org/smartregister/chw/activity/ChildProfileActivity.java:200]: Done

File: opensrp-chw/build.gradle (hunk 3 above)
Comment (githengi) [reply opensrp-chw/build.gradle:59]: Yes lets only have 
```
 buildConfigField "String", "THINKMD_BASE_URL", '"https://thinkmd-v4-app-b-ona.herokuapp.com/"'
 buildConfigField "String", "THINKMD_END_POINT", '"https://fhir-v4-staging-thinkmd-tech.herokuapp.com/fhir"'
```
As the only variables defined via build config

**Pull request closed without merging**